### ISSUE TITLE
How to generate a possible testcase of  an arbitrary function?

### ISSUE THREAD

**Issue by Joney-Yf** [angr/angr#2925]
---

**Comment by Joney-Yf** [comment]
@rhelmot

**Comment by rhelmot** [comment]
Try using depth first search. I believe by default it will take you down a random path.

**Comment by Joney-Yf** [comment]
> Try using depth first search. I believe by default it will take you down a random path.

Yes, I used DFS at first, and then got stuck in an infinite loop, so I chose loopseer, but it still seems to be stuck in one place for a long time, until it is killed.

I even use both DFS and Loopseer, because DFS changes the way of step, the loopseer changes the way of the successor, in theory, it is feasible to use them together, and the program can inherit the characteristics of both. However, the Angr is still stuck in somewhere and cannot move anymore. And I hooked each block, sometimes it still goes into an infinite loop, it seems that the loopseer doesn't work..

Now I use a timelimit to skip the function I can't deal with. But I want to ask if there is a better way to do this?

**Comment by Joney-Yf** [comment]
@rhelmot

**Comment by rhelmot** [comment]
I am not sure. You may want to write a custom exploration technique which prioritizes taking unseen branches in order to hopefully make it out of loops faster.

**Comment by Joney-Yf** [comment]
> I am not sure. You may want to write a custom exploration technique which prioritizes taking unseen branches in order to hopefully make it out of loops faster.

ok， I will try
